packages/gasket-cli/src/index.d.ts

```diff
@@ -1,6 +1,68 @@
 import type { GasketConfigFile, MaybeAsync } from '@gasket/engine';
 import type { PackageManager } from '@gasket/utils';
-import type { Config } from '@oclif/config';
+import type { Command } from 'commander';
+
+// ------ START OF CLI TYPES ------
+export interface Config {
+  bin: Command;
+  root: string;
+  options: Record<string, any>;
+  [key: string]: any;
+}
+
+export interface CLICommand {
+  id: string;
+  description: string;
+  args?: Array<CLICommandArg>;
+  options?: Array<CLICommandOption>;
+  action: (...args: any[]) => MaybeAsync<void>;
+  hidden?: boolean; // Hide from help output
+  default?: boolean; // Default command to run if no command is provided
+}
+
+export interface ProccesedCLICommand {
+  command: Command;
+  hidden: boolean;
+  isDefault: boolean;
+}
+
+// Default cannot be used when required is true
+export type CLICommandArg = {
+  name: string;
+  description: string;
+  required?: true;
+  default?: never;
+} | {
+  name: string;
+  description: string;
+  required?: false;
+  default?: any;
+};
+
+export type ProccesedCLICommandArg = Array<string | boolean>;
+
+export interface CLICommandOption {
+  name: string;
+  description: string;
+  required?: boolean;
+  short?: string;
+  parse?: (value: string) => any;
+  type?: 'string' | 'boolean';
+  conflicts?: Array<string>; // list of option names that cannot be used together
+  hidden?: boolean; // Hide from command help output
+  default?: any; // default value
+}
+
+export interface ProccesedCLICommandOption {
```

Comment:
typo:
```suggestion
export interface ProcessedCLICommandOption {
```

But, CLI* is a bit generic. Could you rename these CLICommand* types and interfaces as GasketCommand*